components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/js/JsAuthenticatedUser.java

```diff
@@ -126,16 +195,16 @@ public boolean hasMember(String name) {
                 return getWrapped().getUserStoreDomain() != null;
             case FrameworkConstants.JSAttributes.JS_TENANT_DOMAIN:
                 return getWrapped().getTenantDomain() != null;
+            case FrameworkConstants.JSAttributes.JS_CLAIMS:
             case FrameworkConstants.JSAttributes.JS_LOCAL_CLAIMS:
-                return idp != null;
             case FrameworkConstants.JSAttributes.JS_REMOTE_CLAIMS:
-                return idp != null && !FrameworkConstants.LOCAL.equals(idp);
+                return true;
             default:
                 return super.hasMember(name);
         }
     }
 
-    protected String[] getLocalRoles() {
+    private String[] getLocalRoles() {
```

Comment:
do we have reason to restrict the scope here?

Reply:
check other places as well

Reply:
This should not be changed.
Addressed here and similar places with 0a520153946f44955800ab413359187671e3f297